runtime/common/src/precompile/stable_asset.rs

```diff
@@ -0,0 +1,672 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2022 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use super::{
+	input::{Input, InputT, Output},
+	target_gas_limit,
+};
+use crate::WeightToGas;
+use frame_support::traits::Get;
+use module_evm::{
+	precompiles::Precompile,
+	runner::state::{PrecompileFailure, PrecompileOutput, PrecompileResult},
+	Context, ExitError, ExitRevert, ExitSucceed,
+};
+use module_support::Erc20InfoMapping;
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use nutsfinance_stable_asset::traits::StableAsset;
+use nutsfinance_stable_asset::WeightInfo;
+use primitives::{Balance, CurrencyId};
+use sp_core::H160;
+use sp_runtime::{traits::Convert, RuntimeDebug};
+use sp_std::{marker::PhantomData, prelude::*};
+
+pub struct StableAssetPrecompile<R>(PhantomData<R>);
+
+#[module_evm_utility_macro::generate_function_selector]
+#[derive(RuntimeDebug, Eq, PartialEq, TryFromPrimitive, IntoPrimitive)]
+#[repr(u32)]
+pub enum Action {
+	GetStableAssetPoolPoolTokens = "getStableAssetPoolTokens(uint32)",
+	GetStableAssetPoolPoolTotalSupply = "getStableAssetPoolPoolTotalSupply(uint32)",
+	GetStableAssetPoolPoolPrecision = "getStableAssetPoolPoolPrecision(uint32)",
+	GetStableAssetPoolPoolMintFee = "getStableAssetPoolPoolMintFee(uint32)",
```

Comment:
Why are there two Pools here? `PoolPool`

Reply:
Ok. I didn't realize that was the case. Gonna fix that.